tests/packages/sharedlib-in-package/mypkg/__init__.py

```diff
@@ -0,0 +1,32 @@
+# SPDX-FileCopyrightText: 2024 The meson-python developers
+#
+# SPDX-License-Identifier: MIT
+
+import os
+
+
+def _enable_sharedlib_loading():
+    """Ensure the `examplelib` shared library on Windows can be loaded
+
+    This shared library is installed alongside this __init__.py file. Due to
+    lack of RPATH support, Windows cannot find shared libraries installed
+    within wheels unless we either amend the DLL search path or pre-load the
+    DLL.
+
+    .. note::
+
+        `os.add_dll_directory` is only available for Python >=3.8, and with
+        the Conda `python` packages only works as advertised for >=3.10.
+        If you require support for older versions, pre-loading the DLL
+        with `ctypes.WinDLL` may be preferred (the SciPy code base has an
+        example of this).
+    """
+    if os.name == "nt":
+        basedir = os.path.dirname(__file__)
+        os.add_dll_directory(basedir)
+
+
+_enable_sharedlib_loading()
```

Comment:
Does this really need to define a function and immediately call it? Given that the setup code is reduced to three lines (that can be condensed to two) I think it is clearer to simply have the code directly at module level. Also, I think that it would be nice to have the code actually shown in the documentation page with a Sphinx code include directive. Then the function doc string can be included in the documentation page (I don't think it is easy to have it rendered nicely in the documentation page otherwise).

Reply:
Makes sense, if it stays at ~2 lines I'll drop the function. `.. literalinclude::` is useful too, will do.

For completeness, `delvewheel` inserts this for external shared libraries that it vendors in:
```python
# start delvewheel patch
def _delvewheel_patch_1_8_2():
    import os
    libs_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), os.pardir, 'numpy.libs'))
    if os.path.isdir(libs_dir):
        os.add_dll_directory(libs_dir)


_delvewheel_patch_1_8_2()
del _delvewheel_patch_1_8_2
# end delvewheel patch
```
The `del` is a nice touch if one wants to keep the namespace very clean.